avogadro/core/graph.cpp

```diff
@@ -57,18 +67,20 @@ bool Graph::isEmpty() const
 void Graph::clear()
 {
   setSize(0);
+  connectedGroup->clear();
 }
 
 size_t Graph::addVertex()
 {
+  connectedGroup->addElement(size() + 1);
```

Comment:
These make me think it's better to just add the ConnectedGroup code directly to the graph class.

Is there a reason you'd have a ConnectedGroup without a Graph? (Seems like "no")
Is there a reason you'd have a Graph without a ConnectedGroup (As written, this is "no")